src/main/webapp/scripts/task.js

```diff
@@ -1,184 +1,161 @@
- */
-function getUserAssignmentButton(task, userID) {
-  const buttonElement = document.createElement('button');
-  buttonElement.setAttribute('type', 'button');
-  if (task.users.includes(userID)) {
-    buttonElement.setAttribute('onclick', 'removeUser()');
-    buttonElement.innerText = 'Remove me from this task';
-  } else {
-    buttonElement.setAttribute('onclick', 'addUser()');
-    buttonElement.innerText = 'Assign me to this task';
-  }
-  return buttonElement;
-}
+  // Add heading to the top of the comment
+  heading = document.createElement('h3');
+  heading.innerText = 'Edit Description';
+  postForm.appendChild(heading);
 
-/**
- * Add user via servlet
- */
-function addUser() {
-  const getParams = new URLSearchParams(location.search);
-  const taskID = getParams.get('taskID');
-  const userID = 1; // Get current userID from cookies?
-  const params = new URLSearchParams('taskID=' + taskID + '&userID=' + userID);
-  console.log('/task-add-user', params);
-  fetch('/task-add-user', {method: 'post', body: params})
-      .then(() => getTaskInfo());
+  // Fill description element with previous description as input
+  descInput = document.createElement('textarea');
+  descInput.setAttribute('type', 'text');
+  descInput.setAttribute('name', 'description');
+  descInput.setAttribute('required', 'true');
+  descInput.innerText = desc;
+  descContainer.appendChild(descInput);
+  postForm.appendChild(descContainer);
+
+  // Fill edit container with Post Edit button
+  editContainer.setAttribute('class', 'inline');
+  postButton = document.createElement('button');
+  postButton.setAttribute('type', 'submit');
+  postButton.setAttribute('class', 'deep-button');
+  postButton.innerText = 'Post';
+  editContainer.appendChild(postButton);
+  postForm.appendChild(editContainer);
+
+  // Create reset container with Reset Edit button
+  // Might be a Discard Changes button instead in the future
+  resetContainer = document.createElement('div');
+  resetContainer.setAttribute('id', 'reset-description-container');
+  resetContainer.setAttribute('class', 'inline');
+  resetButton = document.createElement('button');
+  resetButton.setAttribute('type', 'reset');
+  resetButton.setAttribute('class', 'deep-button');
+  resetButton.innerText = 'Reset';
+  resetContainer.appendChild(resetButton);
+  postForm.appendChild(resetContainer);
 }
 
 /**
- * Add user via servlet
+ * Converts comment into editable comment form.
+ * @param {number} commentID for hidden input.
+ * @param {number} taskID for hidden input.
  */
-function removeUser() {
-  const getParams = new URLSearchParams(location.search);
-  const taskID = getParams.get('taskID');
-  const userID = 1; // Get current userID from cookies?
-  const params = new URLSearchParams('taskID=' + taskID + '&userID=' + userID);
-  console.log('/task-remove-user', params);
-  fetch('/task-remove-user', {method: 'post', body: params})
-      .then(() => getTaskInfo());
+function editComment(commentID, taskID) {
```

Comment:
This is a good addition, allowing to edit the comments by dynamically building the form.  From here you can take it a step further by instead of having the form submit and redirect the page, you can have it send a fetch() call to the servlet instead.  If servlet reports a success, you can update the DOM to display the new text (which you should already have since the user had just typed it in).  If the servlet reports a failure, you can pop  up an alert() (or div lightbox) then revert back to the previous comment (which you can hold on to in a hidden div). 

This'll allow updating multiple comments one after the other without having to refresh the page in between.